difflint.go

```diff
@@ -176,52 +228,35 @@ func isRelativeToCurrentDirectory(path string) bool {
 }
 
 // Check returns the list of unsatisfied rules for the given map of rules.
-func Check(rulesMap map[string][]Rule) (UnsatisfiedRules, error) {
+func Check(rulesMap map[string][]Rule, targetsMap map[string]struct{}) (UnsatisfiedRules, error) {
 	var unsatisfiedRules UnsatisfiedRules
-	for pathnameA, rulesA := range rulesMap {
-	outer:
-		for i, ruleA := range rulesA {
-			// Skip if ruleA is not present or if it has no targets.
-			if len(ruleA.Targets) == 0 || !ruleA.Present {
-				continue
-			}
 
-			for pathnameB, rulesB := range rulesMap {
-			inner:
-				for j, ruleB := range rulesB {
-					// Skip if both rules are present or if ruleA is the same as ruleB.
-					if ruleB.Present || (pathnameA == pathnameB && i == j) {
-						continue
-					}
-
-					// Given that ruleA is present and ruleB is not present, check if ruleA
-					// is satisfied by ruleB.
-					unsatisfiedTargetIndices := make(map[int]struct{})
-					for k, target := range ruleA.Targets {
-						// ruleA is satisfied by ruleB if ruleB matches a target of ruleA.
-						satisfied := target.ID == ruleB.ID && ((target.File == nil && pathnameA == pathnameB) || (*target.File == pathnameB))
-						if satisfied {
-							continue inner
-						}
-
-						// Otherwise, add the target index to the list of unsatisfied targets.
-						unsatisfiedTargetIndices[k] = struct{}{}
-					}
-
-					unsatisfiedRules = append(unsatisfiedRules, UnsatisfiedRule{
-						Rule:               ruleA,
-						UnsatisfiedTargets: unsatisfiedTargetIndices,
-					})
-					continue outer
+	// Check each rule.
+	for _, rules := range rulesMap {
+		for _, rule := range rules {
+			unsatisfiedTargets := make(map[int]struct{}, len(rule.Targets))
+			for i, target := range rule.Targets {
+				key := TargetKey(rule.Hunk.File, target)
+				if _, ok := targetsMap[key]; rule.Present != ok {
```

Comment:
When a rule is present, it means that it is always satisfied. A target is unsatisfied when its rule is not present and the target exists in the `targetsMap`.

Reply:
Resolved by #13.